### PR TITLE
feat: implement workflows add

### PR DESCRIPTION
**Pull request by ereslibre**
This PR introduces the ability to add custom workflows to Rover from URLs, local files, or stdin. Workflows can be stored either locally (project-specific) or globally (user-wide), making workflow management flexible and convenient.

Closes: #345

## Changes

- Added `rover workflows add` command with support for:
  - Local file paths
  - Remote URLs
  - stdin input (using `-` as source)
  - Custom workflow naming via `--name` flag
  - Force global storage via `--global` flag
- Extended `WorkflowStore` class with `saveWorkflow()` method that handles validation, naming conflicts, and storage location
- Added repository initialization checks to commands that require Rover to be initialized (`delete`, `diff`, `inspect`, `iterate`, `logs`, `mcp`, `merge`, `restart`, `shell`, `stop`)
- Added telemetry event tracking for workflow additions (`eventAddWorkflow`)
- Comprehensive test coverage with 581 test cases covering success paths, error handling, edge cases, and stdin input

## Notes

The workflow storage system follows Rover's existing pattern of local vs global configuration:
- **Local workflows** are stored in `.rover/workflows/` within the project
- **Global workflows** are stored in `~/.rover/config/workflows/`
- Workflows are automatically validated against the schema during addition
- Duplicate workflow names within the same store are prevented with clear error messages

### CODE REVIEW

File: packages/core/src/files/workflow-store.ts [hunk 1]
```diff
@@ -55,4 +117,260 @@ export class WorkflowStore {
...
+    } catch (error) {
+      throw new WorkflowStoreError(`Invalid URL: ${url}`, error);
+    }
+  }
+
+  /**
+   * Extract workflow name from local path (filename without extension)
+   */
+  private extractNameFromPath(path: string): string {
+    const fileName = path.split('/').pop() || path.split('\\').pop() || path;
```
Comment (Angelmmiguel) [inline packages/core/src/files/workflow-store.ts]: You can extract the filename using the [`path.basename` Node method](https://nodejs.org/api/path.html#pathbasenamepath-suffix):

```js
import { basename } from 'node:path';

basename(path) // Filename with the extension
```

File: packages/core/src/files/workflow-store.ts [hunk 2]
```diff
@@ -55,4 +117,260 @@ export class WorkflowStore {
...
+      '# Rover Workflow Metadata',
+      `# Source: ${metadata.source}`,
+      `# Imported At: ${metadata.importedAt}`,
+      `# Original Checksum: ${metadata.checksum}`,
+      '#',
+      '',
+    ].join('\n');
+
+    return frontMatter + content;
+  }
```
Comment (Angelmmiguel) [inline packages/core/src/files/workflow-store.ts]: I'm not sure about using a frontmatter, as we need to parse it later on. We might add that information to a special YAML key, but for now let's use it to avoid modifying the original YAML.

**Review by Angelmmiguel — COMMENTED** [review]
Added some last comments before approving 😄

File: packages/core/src/files/workflow-store.ts (hunk 2 above)
Comment (ereslibre) [reply packages/core/src/files/workflow-store.ts]: Yes, right now this is not trying to parse it; just data for information purposes on the workflow itself.

File: packages/cli/src/commands/workflows/add.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import { CLIJsonOutput } from '../../types.js';
+import { exitWithError, exitWithSuccess } from '../../utils/exit.js';
+import { getTelemetry } from '../../lib/telemetry.js';
+import { isJsonMode, setJsonMode } from '../../lib/global-state.js';
+import { readFromStdin } from '../../utils/stdin.js';
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+import { writeFileSync, mkdtempSync, rmSync } from 'node:fs';
+
+interface AddWorkflowCommandOptions {
```
[the remark this answers is not in the document]
Comment (ereslibre) [reply packages/cli/src/commands/workflows/add.ts]: Done!

**Review by Angelmmiguel — APPROVED** [review]
LGTM!
